compiler_opt/rl/data_reader.py

```diff
@@ -45,7 +50,7 @@ def _get_policy_info_parsing_dict(agent_name, action_spec):
 
 
 def _process_parsed_sequence_and_get_policy_info(parsed_sequence, agent_name,
-                                                 action_spec):
+                                                 action_spec, is_greedy):
```

Comment:
what's the use case for is_greedy == True?

Reply:
In distributed training, we have a separate job purely for evaluation and writing stats to tensorboard. That job reads the most recent policy weights from reverb and performs evaluation. Because we are not evaluating using the collect_policy, we don't have the `CategoricalProjectionNetwork_logits` in the output, and so this parsing fails.

To simplify the integration, we could have the distributed learning setup evaluate using the collect_policy, which would make the evaluation a little less faithful but remove this code complexity.

Reply:
the trainer naturally does the `evaluation` of the collect policy during training so i don't see a lot of value in setting up an evaluation job to evaluate the collect policy. 

w.r.t. evaluating the policy during training, how much is the performance difference of the policy and the collect policy?

Reply:
The distributed training infrastructure closely follows the example here: https://github.com/tensorflow/agents/tree/master/tf_agents/experimental/distributed

Evaluation is done separately from the training loop to reduce time spent context switching between the host and device (tpu).

I can remove the is_greedy flag for now, though, just to make the integration more simple.